test/Microsoft.AspNetCore.DataProtection.Extensions.Test/DataProtectionProviderTests.cs

```diff
@@ -57,13 +57,17 @@ public void System_NoKeysDirectoryProvided_UsesDefaultKeysDirectory()
                 var protector = DataProtectionProvider.Create("TestApplication").CreateProtector("purpose");
                 Assert.Equal("payload", protector.Unprotect(protector.Protect("payload")));
 
-                // Step 3: Validate that there's now a single key in the directory and that it's protected using Windows DPAPI.
+                // Step 3: Validate that there's now a single key in the directory
                 var newFileName = Assert.Single(Directory.GetFiles(keysPath));
                 var file = new FileInfo(newFileName);
                 Assert.StartsWith("key-", file.Name, StringComparison.OrdinalIgnoreCase);
                 var fileText = File.ReadAllText(file.FullName);
-                Assert.DoesNotContain("Warning: the key below is in an unencrypted form.", fileText, StringComparison.Ordinal);
-                Assert.Contains("This key is encrypted with Windows DPAPI.", fileText, StringComparison.Ordinal);
+                // On Windows, validate that it's protected using Windows DPAPI.
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    Assert.DoesNotContain("Warning: the key below is in an unencrypted form.", fileText, StringComparison.Ordinal);
```

Comment:
`else { Assert.Contains("Warning: the key below is in an unencrypted form.", fileText, StringComparison.Ordinal); }` ?